File: VB/SingleObjectViewModel.Extensions.vb
Imports System.Linq
Imports System.Collections.Generic
Imports System.ComponentModel
Imports Common.DataModel

Namespace Common.ViewModel
    Partial Public MustInherit Class SingleObjectViewModel(Of TEntity As Class, TPrimaryKey, TUnitOfWork As IUnitOfWork)
        Protected Overrides Function HasValidationErrors() As Boolean
            Dim dataErrorInfo As IDataErrorInfo = TryCast(Entity, IDataErrorInfo)
            If dataErrorInfo Is Nothing Then
                Return MyBase.HasValidationErrors()
            End If
            Return TypeDescriptor.GetProperties(GetType(TEntity)).Cast(Of PropertyDescriptor)().Where(Function(n) Not String.IsNullOrEmpty(dataErrorInfo(n.Name))).Any()
        End Function
    End Class
End Namespace

File: CS/SingleObjectViewModel.Extensions.cs
using System;
using System.Linq;
using System.Collections.Generic;
using System.ComponentModel;
using Scaffolding.ValidationErrors.Common.DataModel;

namespace Scaffolding.ValidationErrors.Common.ViewModel {
    partial class SingleObjectViewModel<TEntity, TPrimaryKey, TUnitOfWork> {

        protected override bool HasValidationErrors() {
            IDataErrorInfo dataErrorInfo = Entity as IDataErrorInfo;
            if(dataErrorInfo == null)
                return base.HasValidationErrors();
            return TypeDescriptor.GetProperties(typeof(TEntity))
                .Cast<PropertyDescriptor>()
                .Where(n => !string.IsNullOrEmpty(dataErrorInfo[n.Name]))
                .Any();
        }
    }
}